spec/forms/state_file/az_retirement_income_form_spec.rb

```diff
@@ -0,0 +1,281 @@
+require "rails_helper"
+
+RSpec.describe StateFile::AzRetirementIncomeForm do
+  let(:intake) { create :state_file_az_intake }
+  let(:form) { described_class.new(intake, params) }
+  before do
+    intake.direct_file_data.fed_taxable_pensions = 1000
+  end
+
+  describe "#valid?" do
+    context "non-numerical values" do
+      let(:params) do
+        {
+          received_military_retirement_payment: "yes",
+          received_military_retirement_payment_amount: "yes i am an amount",
+          primary_received_pension: "yes",
+          primary_received_pension_amount: "&*()$",
+          spouse_received_pension: "yes",
+          spouse_received_pension_amount: "beepboop",
+        }
+      end
+
+      it "is invalid and attaches the correct error" do
+        allow(intake).to receive(:filing_status_mfj?).and_return true
+
+        expect(form).not_to be_valid
+        expect(form.errors[:received_military_retirement_payment_amount]).to include "is not a number"
+        expect(form.errors[:primary_received_pension_amount]).to include "is not a number"
+        expect(form.errors[:spouse_received_pension_amount]).to include "is not a number"
+      end
```

Comment:
Consider replacing with [shoulda_matchers](https://github.com/thoughtbot/shoulda-matchers?tab=readme-ov-file#matchers).

```ruby
it { should validate_numericality_of(:received_military_payment_amount) }
it { should validate_numericality_of(:primary_received_pension_amount) }
it { should validate_numericality_of(:spouse_received_pension_amount) }
```

Reply:
i've never used them! ty!!

Reply:
update on this: turns out using the gem for our form objects isn't that straightforward. i tried it out for a while and slacked kevin about it but will consider it out of scope for this feature

Reply:
Sounds good. Probably something to do with forms being not-quite-models